diff --git a/Services/Registration/classes/class.ilAccountRegistrationGUI.php b/Services/Registration/classes/class.ilAccountRegistrationGUI.php
index c0e22512a9..2d1479d1fd 100644
--- a/Services/Registration/classes/class.ilAccountRegistrationGUI.php
+++ b/Services/Registration/classes/class.ilAccountRegistrationGUI.php
@@ -356,21 +356,24 @@ class ilAccountRegistrationGUI
             $form_valid = false;
         }

+        // BEGIN Patch hide username in registration forms
         // validate username
-        $login_obj = $this->form->getItemByPostVar('username');
-        $login = $this->form->getInput("username");
+        // $login_obj = $this->form->getItemByPostVar('username');
+        // $login = $this->form->getInput("username");
+        $login = "registration_".time();
         if (!ilUtil::isLogin($login)) {
-            $login_obj->setAlert($this->lng->txt("login_invalid"));
+            // $login_obj->setAlert($this->lng->txt("login_invalid"));
             $form_valid = false;
         } elseif (ilObjUser::_loginExists($login)) {
-            $login_obj->setAlert($this->lng->txt("login_exists"));
+            // $login_obj->setAlert($this->lng->txt("login_exists"));
             $form_valid = false;
         } elseif ((int) $ilSetting->get('allow_change_loginname') &&
             (int) $ilSetting->get('reuse_of_loginnames') == 0 &&
             ilObjUser::_doesLoginnameExistInHistory($login)) {
-            $login_obj->setAlert($this->lng->txt('login_exists'));
+            // $login_obj->setAlert($this->lng->txt('login_exists'));
             $form_valid = false;
         }
+        //END Patch

         if (!$form_valid) {
             ilUtil::sendFailure($this->lng->txt('form_input_not_valid'));

diff --git a/Services/User/classes/class.ilUserProfile.php b/Services/User/classes/class.ilUserProfile.php
index 4d32975905..bb8c538399 100755
--- a/Services/User/classes/class.ilUserProfile.php
+++ b/Services/User/classes/class.ilUserProfile.php
@@ -533,7 +533,13 @@ class ilUserProfile
                             $val->setValue($a_user->getLogin());
                         }
                     }
-                    $a_form->addItem($val);
+                    //BEGIN Patch hide username in registration forms
+                    //$a_form->addItem($val);
+                    if(!self::$mode == self::MODE_REGISTRATION)
+                    {
+                        $a_form->addItem($val);
+                    }
+                    //END Patch
                     break;

                 case "text":
