diff --git a/Services/Registration/classes/class.ilAccountRegistrationGUI.php b/Services/Registration/classes/class.ilAccountRegistrationGUI.php
index 44211f4ed0c..f697fc95cc4 100644
--- a/Services/Registration/classes/class.ilAccountRegistrationGUI.php
+++ b/Services/Registration/classes/class.ilAccountRegistrationGUI.php
@@ -315,25 +315,29 @@ class ilAccountRegistrationGUI
             $form_valid = false;
         }
 
+        // BEGIN Patch hide username in registration forms
         // validate username
-        $login_obj = $this->form->getItemByPostVar('username');
-        $login = $this->form->getInput("username");
+        // $login_obj = $this->form->getItemByPostVar('username');
+        // $login = $this->form->getInput("username");
+        $login = "registration_" . time();
+
         if (!ilUtil::isLogin($login)) {
-            $login_obj->setAlert($this->lng->txt("login_invalid"));
+            // $login_obj->setAlert($this->lng->txt("login_invalid"));
             $form_valid = false;
         }
 
         if ($form_valid) {
             if (ilObjUser::_loginExists($login)) {
-                $login_obj->setAlert($this->lng->txt("login_exists"));
+                // $login_obj->setAlert($this->lng->txt("login_exists"));
                 $form_valid = false;
             } elseif ((int) $this->settings->get('allow_change_loginname') &&
                 (int) $this->settings->get('reuse_of_loginnames') === 0 &&
                 ilObjUser::_doesLoginnameExistInHistory($login)) {
-                $login_obj->setAlert($this->lng->txt('login_exists'));
+                // $login_obj->setAlert($this->lng->txt('login_exists'));
                 $form_valid = false;
             }
         }
+        // END Patch
 
         if (!$form_valid) {
             $this->tpl->setOnScreenMessage('failure', $this->lng->txt('form_input_not_valid'));
@@ -379,6 +383,10 @@ class ilAccountRegistrationGUI
             }
         }
 
+        // BEGIN Patch hide username in registration forms
+        $this->userObj->setLogin('login_' . time());
+        // END Patch
+
         $this->userObj->setFullName();
 
         $birthday_obj = $this->form->getItemByPostVar("usr_birthday");
diff --git a/Services/User/classes/Profile/class.ilUserProfile.php b/Services/User/classes/Profile/class.ilUserProfile.php
index f8402976382..1916c6f3b94 100755
--- a/Services/User/classes/Profile/class.ilUserProfile.php
+++ b/Services/User/classes/Profile/class.ilUserProfile.php
@@ -67,6 +67,11 @@ class ilUserProfile
                 in_array($f, $this->skip_fields)) {
                 continue;
             }
+            // BEGIN PATCH AutogenerateUsername
+            if ($f == 'username') {
+                continue;
+            }
+            // END PATCH AutogenerateUsernam
             $fields[$f] = $p;
         }
         return $fields;
@@ -451,8 +456,13 @@ class ilUserProfile
             $login_input->setSize(255);
             $login_input->setRequired(true);
         }
-
-        if ($user !== null) {
+        // BEGIN Patch hide username in registration forms
+        if (
+            $user !== null &&
+            !$this->mode == self::MODE_REGISTRATION
+        ) {
+        //if ($user !== null) {
+        // END Patch
             $login_input->setValue($user->getLogin());
         }
         return $login_input;
